### DISCUSSION TITLE
Can't redirect after submitting the form (production build only)

### DISCUSSION THREAD

**Discussion by bartekczyz** [vercel/next.js#52324] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 22.5.0: Thu Jun  8 22:22:20 PDT 2023; root:xnu-8796.121.3~7/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.13.2
      npm: 8.1.2
      Yarn: 1.22.19
      pnpm: 7.33.1
    Relevant Packages:
      next: 13.4.8-canary.14
      eslint-config-next: 13.4.8-canary.14
      react: 18.2.0
      react-dom: 18.2.0
      typescript: 5.1.6
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Middleware / Edge (API routes, runtime), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue or a replay of the bug

https://github.com/itzaymvn/aymvn.tech

### To Reproduce

[This function](https://github.com/itzAymvn/aymvn.tech/blob/master/app/(admin)/login/page.js#L11) is supposed to [redirect the user to the `/dashboard` route](https://github.com/itzAymvn/aymvn.tech/blob/master/app/(admin)/login/page.js#L58) if the sign in is successful, It works like that when `npm run dev` and everything perfect. But when I run the build (`npm run build` & `npm run start`) it just logs both "Before redirect" and "After redirect" but doesn't actually redirect.

### Describe the Bug

It happens only on production build: can't redirect after submitting the form. No matter if I use `window.location.replace` or `useRouter`

### Expected Behavior

I should be redirected after submitting the form. It actually happens when I run the app locally, using `npm run dev`

**Comment by bartekczyz** [comment]
This can be fixed by running `router.refresh()` just before redirecting, see more https://github.com/vercel/next.js/discussions/51782#discussioncomment-6470680

**Reply by xiliumz** [reply]
do you have router.prefetch in your code before?

**Reply by bartekczyz** [reply]
No, I don't
